test/react/FunctionalComponents/simple-22.js

```diff
@@ -0,0 +1,25 @@
+var React = require("react");
+
+function Child() {
+  return <div>This should be inlined</div>;
+}
+
+function Child2() {
+  return <span>This should be inlined too</span>;
+}
+
+function App(props) {
+  var a = props.x ? null : <Child2 />;
+  return a && <Child />;
```

Comment:
I don't understand what this is supposed to test. What does `<Child2 /> && <Child />` mean?

Judging by the issue description I would assume the test would be something like

```js
function App(props) {
  if (props.neverHappens) {
    return <Bad />
  }
  return null;
}

function Bad() {
  return {}; // Invalid
}
```

and then the test should check we still render successfully.

Reply:
Will add this another test. The original tests from taken from my issue, test `simple-23` fail with Herman's PR applied and `simple-22` doesn't fully inline without Herman's PR.